### PR TITLE
Add rk support

### PR DESCRIPTION
**Pull request by oliviazhang0809**
support adding routing key for catalyst services

### CODE REVIEW

File: runtime/tchannel_client.go [hunk 1]
```diff
@@ -154,10 +157,13 @@ func (c *tchannelClient) Call(ctx context.Context, thriftService, methodName str
 	retryOpts := &tchannel.RetryOptions{
 		TimeoutPerAttempt: c.timeoutPerAttempt,
 	}
-	ctx, cancel := tchannel.NewContextBuilder(c.timeout).
+	ctxBuilder := tchannel.NewContextBuilder(c.timeout).
 		SetParentContext(ctx).
-		SetRetryOptions(retryOpts).
-		Build()
+		SetRetryOptions(retryOpts)
+	if c.routingKey != nil {
```
Comment (zhenghuiwang) [inline runtime/tchannel_client.go]: Nit: It seems to be simpler to use `string` as type for `routingKey` and check if it is not empty string.

**Comment by Raynos** [comment]
Will also need code changes in template that calls `NewTChannelClient()` ?

File: runtime/static_config.go [hunk 2]
```diff
@@ -206,6 +206,21 @@ func (conf *StaticConfig) MustGetInt(key string) int64 {
 	panic(errors.Errorf("Key (%s) not available", key))
 }
 
+// ContainsKey returns the value as a string or panics.
+func (conf *StaticConfig) ContainsKey(key string) bool {
+	if conf.destroyed {
+		panic(errors.Errorf("Cannot get(%s) because destroyed", key))
```
Comment (Raynos) [inline runtime/static_config.go]: Cannot containsKey(%s)

File: runtime/tchannel_client.go (hunk 1 above)
Comment (ChuntaoLu) [reply runtime/tchannel_client.go]: +1, seems like `string` type is sufficient

**Comment by Raynos** [comment]
@oliviazhang0809 

Travis says make cover failed ( not enough coverage ).

You can run `make cover ; make view-istanbul` to run the tests with coverage and view lack of coverage.

**Pull request merged**